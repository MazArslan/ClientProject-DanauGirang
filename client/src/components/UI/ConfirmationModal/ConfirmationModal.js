import React from "react";
import propTypes from "prop-types";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";

/**
 * Modal asking the user to confirm their action
 * @param { modal, toggle, onSubmit } props
 */
const confirmationModal = props => {
  const { modal, toggle, onSubmit } = props;
  return (
    <Modal style={{ zIndex: 2000 }} isOpen={modal} toggle={toggle}>
      <ModalHeader toggle={toggle}>Submission confirmation</ModalHeader>
      <ModalBody>Are you sure you want to submit?</ModalBody>
      <ModalFooter>
        <Button className="cancel-button" color="secondary" onClick={toggle}>
          Cancel
        </Button>
        <Button className="submit-button" color="success" onClick={onSubmit}>
          Yes, Submit
        </Button>
      </ModalFooter>
    </Modal>
  );
};

confirmationModal.propTypes = {
  modal: propTypes.bool.isRequired,
  toggle: propTypes.func.isRequired,
  onSubmit: propTypes.func.isRequired
};

export default confirmationModal;
