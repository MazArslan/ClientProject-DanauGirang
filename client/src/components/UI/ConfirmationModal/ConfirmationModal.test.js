import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import ConfirmationModal from "./ConfirmationModal";

Enzyme.configure({ adapter: new Adapter() });

describe("<ConfirmationModal/>", () => {
  const onActionMock = jest.fn();
  const props = {
    modal: true,
    toggle: onActionMock,
    onSubmit: onActionMock
  };
  const container = shallow(<ConfirmationModal {...props} />);

  describe("component rendering", () => {
    it("should render correctly", () => {
      expect(container.exists()).toBe(true);
      expect(toJson(container)).toMatchSnapshot();
    });

    it("should render modal header", () => {
      const textHeading = "Submission confirmation";
      expect(container.find("ModalHeader").props().children).toEqual(
        textHeading
      );
    });

    it("should render modal body", () => {
      const textBody = "Are you sure you want to submit?";
      expect(container.find("ModalBody").props().children).toEqual(textBody);
    });

    it("should render cancel button", () => {
      expect(
        container
          .find("Button")
          .at(0)
          .props().className
      ).toEqual("cancel-button");
    });

    it("should render submit button", () => {
      expect(
        container
          .find("Button")
          .at(1)
          .props().className
      ).toEqual("submit-button");
    });
  });
});
