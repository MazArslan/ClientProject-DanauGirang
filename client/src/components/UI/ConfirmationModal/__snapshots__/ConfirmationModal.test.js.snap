// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationModal/> component rendering should render correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  returnFocusAfterClose={true}
  role="dialog"
  scrollable={false}
  style={
    Object {
      "zIndex": 2000,
    }
  }
  toggle={[MockFunction]}
  unmountOnClose={true}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    toggle={[MockFunction]}
    wrapTag="div"
  >
    Submission confirmation
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    Are you sure you want to submit?
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      className="cancel-button"
      color="secondary"
      onClick={[MockFunction]}
      tag="button"
    >
      Cancel
    </Button>
    <Button
      className="submit-button"
      color="success"
      onClick={[MockFunction]}
      tag="button"
    >
      Yes, Submit
    </Button>
  </ModalFooter>
</Modal>
`;
